engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/HashCodeNode.java

```diff
@@ -111,29 +129,80 @@ long hashCodeForAtomConstructor(AtomConstructor atomConstructor) {
     return System.identityHashCode(atomConstructor);
   }
 
-  /** How many {@link HashCodeAnyNode} nodes should be created for fields in atoms. */
+  @Specialization
+  long hashCodeForUnresolvedSymbol(UnresolvedSymbol unresolvedSymbol,
+      @Cached HashCodeNode hashCodeNode) {
+    long nameHash = hashCodeNode.execute(unresolvedSymbol.getName());
+    long scopeHash = hashCodeNode.execute(unresolvedSymbol.getScope());
+    return Objects.hashCode(nameHash, scopeHash);
+  }
+
+  @Specialization
+  long hashCodeForUnresolvedConversion(UnresolvedConversion unresolvedConversion,
+      @CachedLibrary(limit = "5") InteropLibrary interop) {
+    return hashCodeForModuleScope(unresolvedConversion.getScope(), interop);
+  }
+
+  @Specialization
+  long hashCodeForModuleScope(ModuleScope moduleScope,
+      @CachedLibrary(limit = "5") InteropLibrary interop) {
+    return hashCodeForModule(moduleScope.getModule(), interop);
+  }
+
+  @Specialization
+  @TruffleBoundary
+  long hashCodeForModule(Module module,
+      @CachedLibrary(limit = "5") InteropLibrary interop) {
+    return hashCodeForString(module.toString(), interop);
```

Comment:
While `==` of `Module` & co. is very likely Java `==`, computing `hash` for these objects from their name makes the hash more stable and transferable between two runs of the engine. Using `module.toString()` to compute the hash is probably good idea.